x/amm/keeper/source.go

```diff
@@ -33,12 +33,19 @@ func (k OrderSource) ConstructMemOrderBookSide(
 	if !found {
 		return nil // no pool found
 	}
+	maxPriceRatio := k.exchangeKeeper.GetMaxOrderPriceRatio(ctx)
+	poolState := k.MustGetPoolState(ctx, pool.Id)
+	minPrice, maxPrice := exchangetypes.OrderPriceLimit(poolState.CurrentPrice, maxPriceRatio)
 
 	reserveAddr := pool.MustGetReserveAddress()
 	accQty := utils.ZeroDec
 	accQuote := utils.ZeroDec
 	numPriceLevels := 0
 	k.IteratePoolOrders(ctx, pool, opts.IsBuy, func(price, qty, openQty sdk.Dec) (stop bool) {
+		if opts.IsBuy && price.LT(minPrice) ||
+			!opts.IsBuy && price.GT(maxPrice) {
```

Comment:
It could be seem ambiguous, How about this
```suggestion
		if (opts.IsBuy && price.LT(minPrice)) ||
			(!opts.IsBuy && price.GT(maxPrice)) {
```

And the rules are different from [here](https://github.com/crescent-network/crescent/blob/main/x/exchange/keeper/order.go#L140-L145), is it okay?

Reply:
I'll commit the suggestion.

> And the rules are different from [here](https://github.com/crescent-network/crescent/blob/main/x/exchange/keeper/order.go#L140-L145), is it okay?

Yes, the rule is different.